test/algorithms/test_effective_dimension.py

```diff
@@ -0,0 +1,210 @@
+
+    def setUp(self):
+        super().setUp()
+
+        # fix seeds
+        algorithm_globals.random_seed = 12345
+        qi_sv = QuantumInstance(
+            Aer.get_backend("aer_simulator_statevector"),
+            seed_simulator=algorithm_globals.random_seed,
+            seed_transpiler=algorithm_globals.random_seed,
+        )
+
+        # set up qnns
+        num_qubits = 3
+        feature_map = ZFeatureMap(feature_dimension=num_qubits, reps=1)
+        ansatz = RealAmplitudes(num_qubits, reps=1)
+
+        # CircuitQNNs
+        qc = QuantumCircuit(num_qubits)
+        qc.append(feature_map, range(num_qubits))
+        qc.append(ansatz, range(num_qubits))
+
+        def parity(x):
+            return f"{x:b}".count("1") % 2
+
+        self.circuit_qnn_1 = CircuitQNN(
+            qc,
+            input_params=feature_map.parameters,
+            weight_params=ansatz.parameters,
+            interpret=parity,
+            output_shape=2,
+            sparse=False,
+            quantum_instance=qi_sv,
+        )
+
+        # qnn2 for checking result without parity with
+        self.circuit_qnn_2 = CircuitQNN(
+            qc,
+            input_params=feature_map.parameters,
+            weight_params=ansatz.parameters,
+            sparse=False,
+            quantum_instance=qi_sv,
+        )
+
+        # OpflowQNN
+        observable = PauliSumOp.from_list([("Z" * num_qubits, 1)])
+        self.opflow_qnn = TwoLayerQNN(
+            num_qubits,
+            feature_map=feature_map,
+            ansatz=ansatz,
+            observable=observable,
+            quantum_instance=qi_sv,
+        )
+
+        # define sample numbers
+        self.n_list = [5000, 8000, 10000, 40000, 60000, 100000, 150000, 200000, 500000, 1000000]
+        self.n = 5000
+
+        # define results
+        self.result1 = 4.62355185  # 3,10,10
+        self.result2 = 1.39529449  # 3,1,1
+        self.result3 = 4.92825035  # 3,10,1
+        self.result4 = 5.93064172  # circuitqnn2 with 3,10,10
+
+    @data(
+        # num_inputs, num_params
+        ("circuit1", 10, 10),
+        ("circuit1", 1, 1),
+        ("circuit1", 10, 1),
+        ("circuit2", 10, 10),
+    )
+    def test_alg_results(self, config):
+        """Test that the algorithm results match the original code's."""
+
+        qnn_name, num_inputs, num_params = config
+
+        if qnn_name == "circuit2":
+            qnn = self.circuit_qnn_2
+            result = self.result4
+        else:
+            qnn = self.circuit_qnn_1
+            if num_inputs == 1:
+                result = self.result2
+            elif num_params == 10:
+                result = self.result1
+            else:
+                result = self.result3
+
+        global_ed = EffectiveDimension(
+            qnn=qnn, num_params=num_params, num_inputs=num_inputs, fix_seed=True
+        )
+
+        effdim = global_ed.get_eff_dim(self.n)
+
+        self.assertAlmostEqual(effdim, result, 5)
```

Comment:
Would it be easier to read 4 independent tests instead of this sophisticated `if` statement?

Reply:
Or pass the result in with each tuple in the data so its one test with the inputs and expected result passed in.